spanner/benchmarks/benchmarks.go

```diff
@@ -0,0 +1,156 @@
+)
+
+const (
+	selectQuery  = "SELECT ID from EMPLOYEES WHERE ID = @p1"
+	totalRecords = 100000
+)
+
+func main() {
+	ctx := context.Background()
+
+	project := os.Getenv("SPANNER_CLIENT_BENCHMARK_GOOGLE_CLOUD_PROJECT")
+	instance := os.Getenv("SPANNER_CLIENT_BENCHMARK_SPANNER_INSTANCE")
+	database := os.Getenv("SPANNER_CLIENT_BENCHMARK_SPANNER_DATABASE")
+
+	if project == "" || instance == "" || database == "" {
+		fmt.Println(`You must set all the environment variables SPANNER_CLIENT_BENCHMARK_GOOGLE_CLOUD_PROJECT, 
+			SPANNER_CLIENT_BENCHMARK_SPANNER_INSTANCE and SPANNER_CLIENT_BENCHMARK_SPANNER_DATABASE`)
+		return
+	}
+
+	if len(os.Args) < 4 {
+		fmt.Println("Please set warm up time, execution time and wait between requests in the command line")
+		return
+	}
+
+	warmupTime, _ := strconv.ParseInt(os.Args[1], 10, 8)          // in minutes
+	executionTime, _ := strconv.ParseInt(os.Args[2], 10, 8)       // in minutes
+	waitBetweenRequests, _ := strconv.ParseInt(os.Args[3], 10, 8) // in milliseconds
+
+	db := fmt.Sprintf("projects/%v/instances/%v/databases/%v", project, instance, database)
+
+	client, err := spanner.NewClientWithConfig(ctx, db, spanner.ClientConfig{})
+	if err != nil {
+		return
+	}
+
+	err = warmUp(ctx, client, warmupTime)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+
+	latencies, err := runBenchMark(ctx, client, executionTime, waitBetweenRequests)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+	slices.Sort(latencies)
+
+	fmt.Printf("\nResults\np50 %v\n p95 %v\n p99 %v\n", percentiles(50, latencies),
+		percentiles(95, latencies), percentiles(99, latencies))
+}
+
+func warmUp(ctx context.Context, client *spanner.Client, warmupTime int64) error {
+	endTime := time.Now().Local().Add(time.Minute * time.Duration(warmupTime))
+
+	go runTimer(endTime, "Remaining warmup time")
+	for {
+		if time.Now().Local().After(endTime) {
+			break
+		}
+		_, err := executeQuery(ctx, client)
+		if err != nil {
+			fmt.Println(err)
+			return err
+		}
+	}
+	return nil
+}
+
+func runBenchMark(ctx context.Context, client *spanner.Client, executionTime int64, waitBetweenRequests int64) ([]int64, error) {
+	endTime := time.Now().Local().Add(time.Minute * time.Duration(executionTime))
+
+	go runTimer(endTime, "Remaining operation time")
+	var durations []int64
+	for {
+		if time.Now().Local().After(endTime) {
+			break
+		}
+		duration, err := executeQuery(ctx, client)
+		if err != nil {
+			fmt.Println(err)
+			return make([]int64, 0), err
+		}
+		durations = append(durations, duration)
+		time.Sleep(time.Millisecond * time.Duration(waitBetweenRequests))
+	}
+
+	return durations, nil
+}
+
+func executeQuery(ctx context.Context, client *spanner.Client) (int64, error) {
+	start := time.Now()
+
+	iter := client.Single().Query(ctx, spanner.Statement{SQL: selectQuery, Params: map[string]interface{}{
+		"p1": generateUniqueID(),
+	}})
+	for {
+		_, err := iter.Next()
+		if errors.Is(err, iterator.Done) {
+			break
+		}
+		if err != nil {
```

Comment:
Note that this means that you are never decoding the results from protobuf objects to Go objects. That is OK, but should be taken into account when compared to for example the Java results, if the latter does decode (which it does by default)